eva-server/src/test/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2IntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+import com.lordofthejars.nosqlunit.mongodb.MongoDbRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import uk.ac.ebi.eva.commons.core.models.Annotation;
+import uk.ac.ebi.eva.commons.core.models.ws.VariantSourceEntryWithSampleNames;
+import uk.ac.ebi.eva.commons.core.models.ws.VariantWithSamplesAndAnnotation;
+import uk.ac.ebi.eva.commons.mongodb.services.VariantWithSamplesAndAnnotationsService;
+import uk.ac.ebi.eva.lib.Profiles;
+import uk.ac.ebi.eva.lib.utils.QueryResponse;
+import uk.ac.ebi.eva.lib.utils.QueryResult;
+import uk.ac.ebi.eva.server.configuration.MongoRepositoryTestConfiguration;
+
+import java.net.URISyntaxException;
+import java.util.List;
+
+import static com.lordofthejars.nosqlunit.mongodb.MongoDbRule.MongoDbRuleBuilder.newMongoDbRule;
+import static org.junit.Assert.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Import(MongoRepositoryTestConfiguration.class)
+@UsingDataSet(locations = {
+        "/test-data/variants.json",
+        "/test-data/files.json",
+        "/test-data/annotations.json",
+        "/test-data/annotation_metadata.json"
+})
+@ActiveProfiles(Profiles.TEST_MONGO_FACTORY)
+public class VariantWSServerV2IntegrationTest {
+
+    private static final String TEST_DB = "test-db";
+    @Rule
+    public MongoDbRule mongoDbRule = newMongoDbRule().defaultSpringMongoDb(TEST_DB);
+    @Autowired
+    MongoDbFactory mongoDbFactory;
+    @Autowired
+    private ApplicationContext applicationContext;
+    @Autowired
+    private TestRestTemplate restTemplate;
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @Before
+    public void setUp() throws Exception { }
+
+    @Test
+    public void rootTestGetVariantsByVariantId() throws URISyntaxException {
+        String url = "/v2/variants/20:60100:A:T?species=mmusculus&assembly=grcm38";
+        List<VariantWithSamplesAndAnnotation> variantWithSamplesAndAnnotations = WSTestHelpers.testRestTemplateHelper(
+                url,restTemplate);
+        assertTrue(variantWithSamplesAndAnnotations.size() > 0);
+        assertTrue(variantWithSamplesAndAnnotations.get(0).getSourceEntries().size() == 0);
```

Comment:
This looks more natural to me, either that or `assertEquals(0, ...)`.
```suggestion
        assertTrue(variantWithSamplesAndAnnotations.get(0).getSourceEntries().isEmpty());
```